Assets/Mirror/Runtime/SyncSet.cs

```diff
@@ -41,6 +41,14 @@ protected SyncSet(ISet<T> objects)
             this.objects = objects;
         }
 
+        public void Reset()
+        {
+            IsReadOnly = false;
+            changes.Clear();
+            changesAhead = 0;
+            objects.Clear();
```

Comment:
Not quite correct,  when a user creates a synclist,  he can initialize it with some elements.  

I don't really have a good solution though

Reply:
The reset needs to do the same as creating a new instance, so it just needs to reset these fields to their default values.

I'm not sure what you mean by "initialize it with some elements." none of the constructors let people pass in items when the list is created.